src/pyfaf/ureport.py

```diff
@@ -537,11 +550,51 @@ def is_known(ureport, db, return_report=False, opsysrelease_id=None):
     problemplugin = problemtypes[ureport["problem"]["type"]]
     report_hash = problemplugin.hash_ureport(ureport["problem"])
 
-    report = get_report_by_hash(db, report_hash)
+    known_type = []
+
+    # Split allowed types from config
+    if config.has_key('ureport.known') \
+        and config['ureport.known'].strip() != "":
+        known_type = config['ureport.known'].strip().split(" ")
+
+    if len(known_type) > 0 and not valid_known_type(known_type):
+        return None
+
+    report_os = None
+
+    if 'EQUAL_UREPORT_EXISTS' in known_type:
+        report_os = ureport["os"]
+
+    reports = db.session.query(Report).all()
+
+    report = find_report(db, report_hash, report_os=report_os)
 
     if report is None:
         return None
-    if get_reportbz(db, report.id, opsysrelease_id).first() is not None:
+
+    if 'EQUAL_UREPORT_EXISTS' in known_type and report is not None:
+        if return_report:
+            return report
+        return True
+
+    elif 'BUG_OS_MINOR_VERSION' in known_type and report is not None \
+        and get_reportbz(db, report.id, opsysrelease_id).first() is not None:
```

Comment:
It's better to put multi-line conditions into parenthesis instead of '\'. You're also missing new line after condition, pep8 warns about this.